### ISSUE TITLE
Add missing indicator instead of dropping columns above null fraction threshold

### ISSUE THREAD

**Issue by jeromedockes** [skrub-data/skrub#1714]
### Problem Description

In the Cleaner and TableVectorizer we can set a `drop_null_fraction`, columns that have more nulls than this are dropped. We could consider, rather than dropping, replacing with a boolean column that indicates the positions of nulls as done by scikit-learn's [MissingIndicator](https://scikit-learn.org/stable/modules/generated/sklearn.impute.MissingIndicator.html) . it still acheives the goal of not wasting many feature dimensions on such columns but retains the information that the field was present or not, which is often informative. for example if there is a column like "blood pressure medication" we may not want to spend lots of dimension to one-hot or string-encode it but retain the information that the patient is taking such a medication or not 

### Feature Description

instead of dropping columns with too many nulls, they would be replaced by their missingness indicator. (columns that are entirely null should still be dropped, which will happen by default because we drop all constant columns and the missingness mask would be constant)

### Alternative Solutions

keep the current behavior (dropping), and users can use the missing indicator by themselves where relevant

**Comment by rcap107** [comment]
The code part of this issue has been addressed in #1881, but the documentation is still missing an example on how selectors can be used to apply the missing indicator to columns that include many nulls thanks to this selector: we should add one such example before closing this issue.

**Comment by rcap107** [comment]
Rough idea of what the example could look like:

```python
# %%
import pandas as pd
from sklearn.impute import MissingIndicator, SimpleImputer
from skrub import ApplyToCols
from sklearn.pipeline import make_pipeline
import skrub.selectors as s

# Generate a DataFrame with missing values
df = pd.DataFrame(
    {
        "missing": [1, 2, None, 4, None, 6, 7, None, 9, 10],
        "value": [10.5, 20.3, 15.8, 25.1, 30.9, 18.2, 22.7, 28.4, 16.6, 24.5],
    }
)

print(df)
# %%
indicator = ApplyToCols(
    MissingIndicator(), cols=s.has_nulls(), keep_original=True
)
imputer = ApplyToCols(
    SimpleImputer(), cols=s.has_nulls()
)
pipeline = make_pipeline(indicator, imputer)
result = pipeline.fit_transform(df)
print(result)
# %%
```
which gives

```
   value  missingindicator_missing    missing
0   10.5                     False   1.000000
1   20.3                     False   2.000000
2   15.8                      True   5.571429
3   25.1                     False   4.000000
4   30.9                      True   5.571429
5   18.2                     False   6.000000
6   22.7                     False   7.000000
7   28.4                      True   5.571429
8   16.6                     False   9.000000
9   24.5                     False  10.000000
```
